onnxruntime/test/platform/threadpool_test.cc

```diff
@@ -137,6 +137,30 @@ void TestBurstScheduling(const std::string& name, int num_tasks) {
   }
 }
 
+void TestPoolCreation(const std::string&, int iter) {
+  // Test creating and destroying thread pools.  This can be used with Valgrind to help
+  // check for memory leaks related to the initialization and clean-up code.  For instance
+  //
+  //  valgrind --leak-check=full ./onnxruntime_test_all --gtest_filter=ThreadPoolTest.TestPoolCreation_10Iter
+  //
+  // We create #iter thread pools, and within each of them run a loop of #per_iter steps.
+  std::atomic<std::ptrdiff_t> ctr{0};
+  constexpr std::ptrdiff_t per_iter = 1024;
+  constexpr int num_threads = 4;
+  for (auto i = 0; i < iter; i++) {
```

Comment:
I think we can't do this. Because Deleting a thread pool is very very expensive on our CI build machines.